rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -245,6 +247,38 @@ public Topology getTopology() {
         }
     }
 
+    private NodeSet getNodeSetWithExtraNodes(LinkedHashSet<Node> nodes, int number) {
+        LinkedHashSet<Node> main = new LinkedHashSet<Node>();
+        LinkedHashSet<Node> extra = new LinkedHashSet<Node>();
+        int i = 0;
+        for (Node n : nodes) {
+            if (i < number) {
+                main.add(n);
+            } else {
+                extra.add(n);
+            }
+            i++;
+        }
+        NodeSet result = new NodeSet(main);
+        result.setExtraNodes(extra);
+        return result;
+    }
+
+    private LinkedHashSet<Node> subListLHS(LinkedHashSet<Node> lhs, int begin, int end) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "lhs" object should be an interface such as "Set" rather than the implementation "LinkedHashSet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The return type of this method should be an interface such as "Set" rather than the implementation "LinkedHashSet". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)